pkg/operator/logging/logging.go

```diff
@@ -94,8 +105,44 @@ func WithCommit(logger *zap.SugaredLogger) *zap.SugaredLogger {
 	return logger.With(zap.String(logkey.Commit, revision))
 }
 
-func defaultLogger() *zap.SugaredLogger {
-	return WithCommit(lo.Must(DefaultZapConfig().Build()).Sugar())
+func defaultLogger(ctx context.Context, component string) *zap.SugaredLogger {
+	cfg := DefaultZapConfig()
+	if l := injection.GetOptions(ctx).LogLevel; l != "" {
+		// Webhook log level can only be configured directly through the zap-config
+		// Webhooks are deprecated, so support for changing their log level is also deprecated
+		if component != "webhook" {
+			cfg.Level = lo.Must(zap.ParseAtomicLevel(l))
+		}
+	}
+	return WithCommit(lo.Must(cfg.Build()).Sugar()).Named(component)
+}
+
+func loggerFromFile(ctx context.Context, component string) *zap.SugaredLogger {
+	raw, err := os.ReadFile(loggerCfgFilePath)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil
+		}
+		log.Fatalf("retrieving logging configuration file from %q", loggerCfgFilePath)
+	}
+	cfg := DefaultZapConfig()
+	lo.Must0(json.Unmarshal(raw, &cfg))
+
+	raw, err = os.ReadFile(loggerCfgDir + fmt.Sprintf("/loglevel.%s", component))
+	if err != nil && !os.IsNotExist(err) {
+		log.Fatalf("retrieving logging controller log level file from %q", loggerCfgDir+fmt.Sprintf("/loglevel.%s", component))
```

Comment:
If the log level doesn't exist, should we just return and fall back to the default?

Reply:
Yes, I think so. A user is allowed not to have this file and (if they don't have it) then we should just use the default. I _do_ still want to catch errors that aren't related to notFound like transient errors with the filesystem.